pelican/tests/test_generators.py

```diff
@@ -174,6 +176,25 @@ def test_bad_prefix(self):
                           '!UNKNOWN/authors')
 
 
+class TestParseErrors(LoggedTestCase):
+
+    def test(self):
+        # Verify that just an error is printed and the application doesn't
+        # abort, exit or something. Testing here and not in the reST reader
+        # test as only the page/article generators have the final exception
+        # catching routine.
```

Comment:
Given that you expect an `Exception` and nothing else, you can test using `assertRaises`.

Reply:
I wanted to put it here so it actually verifies that a logger error comes out (and not that the exception is uncaught, like it was before).

Using `assertRaises()` in `test_readers.py` was my first attempt, but that ultimately felt "not good enough", as it doesn't test anything about the exception-to-error conversion in the generators and so it could still result in the exception causing an abort in case the code in `generators.py` changes in some way.

(But if you think `assertRaises()` in `test_readers.py` is better, sure, I can go back to that.)

Reply:
Ah, OK. The comment made it look like you *had to* test like this. This looks somewhat out of place currently :).

How about this:
* One test explicitly in `RstReaderTest`
* And... 
    * We have a few "full" `PageGenerator` tests down below (Like [this](https://github.com/getpelican/pelican/blob/c3aaa97a13dae32aa0d13e78df94ccf25045adb1/pelican/tests/test_generators.py#L635)). Place the page somewhere this can pick up and assert for the error log in addition to others.
    * Or... alternatively. You move this to `test_pelican`, which does similar log counts already.

Reply:
Done :)